src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AccountingPoliciesService.java

```diff
@@ -0,0 +1,180 @@
+
+            repository.insert(entity);
+        } catch (DuplicateKeyException e) {
+
+            LOGGER.errorRequest(request, e, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR);
+        } catch (MongoException e) {
+
+            DataException dataException = new DataException("Failed to insert " + ResourceName.ACCOUNTING_POLICIES.getName(), e);
+            LOGGER.errorRequest(request, dataException, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            throw dataException;
+        }
+
+        smallFullService
+                .addLink(companyAccountsId, SmallFullLinkType.ACCOUNTING_POLICY_NOTE,
+                        getSelfLinkFromAccountingPolicy(entity), request);
+
+        return new ResponseObject<>(ResponseStatus.CREATED, rest);
+    }
+
+    @Override
+    public ResponseObject<AccountingPolicies> update(AccountingPolicies rest, Transaction transaction,
+                                                     String companyAccountsId, HttpServletRequest request)
+                                                    throws DataException {
+
+        setMetadataOnRestObject(rest, transaction, companyAccountsId);
+
+        AccountingPoliciesEntity entity = transformer.transform(rest);
+        entity.setId(generateID(companyAccountsId));
+
+        try {
+
+            repository.save(entity);
+        } catch (MongoException e) {
+
+            DataException dataException = new DataException("Failed to update " + ResourceName.ACCOUNTING_POLICIES.getName(), e);
+            LOGGER.errorRequest(request, dataException, getDebugMap(transaction, companyAccountsId, entity.getId()));
+
+            throw dataException;
+        }
+
+        return new ResponseObject<>(ResponseStatus.UPDATED, rest);
+    }
+
+    @Override
+    public ResponseObject<AccountingPolicies> findById(String id, HttpServletRequest request) throws DataException {
+
+        AccountingPoliciesEntity entity;
+
+        try {
+
+            entity = repository.findById(id).orElse(null);
+        } catch (MongoException e) {
+
+            final Map<String, Object> debugMap = new HashMap<>();
+            debugMap.put("id", id);
+
+            DataException dataException = new DataException("Failed to find AccountingPolicies", e);
+            LOGGER.errorRequest(request, dataException, debugMap);
+
+            throw dataException;
+        }
+
+        if (entity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+
+        return new ResponseObject<>(ResponseStatus.FOUND, transformer.transform(entity));
+    }
+
+    @Override
+    public String generateID(String companyAccountId) {
+
+        return keyIdGenerator.generate(companyAccountId + "-" + ResourceName.ACCOUNTING_POLICIES.getName());
+    }
+
+    private Map<String, Object> getDebugMap(Transaction transaction, String companyAccountsId, String id) {
```

Comment:
Could this be added as a default method to the interface so it can be reused between services?